recipes/giflib/5.2.x/conanfile.py

```diff
@@ -0,0 +1,55 @@
+from conans import CMake, ConanFile, tools
+import os
+
+class GiflibConan(ConanFile):
+    name = "giflib"
+    description = "A library and utilities for reading and writing GIF images."
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    homepage = "http://giflib.sourceforge.net"
+    topics = ("conan", "giflib", "image", "multimedia", "format", "graphics")
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("%s-%s" % (self.name, self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
```

Comment:
Shouldn't patching be done in `source`?

Reply:
No. Source folder should be untouched. Why? Any change in the recipe will require a new download, which means more time. Keep all patches on build folder, which is a "build cache".